### ISSUE TITLE
Download KubeConfig fails to download when MCM feature flag is disabled

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#33687]
**Rancher Server Setup**
- Rancher version: master-head
- Installation option (Docker install/Helm Chart): Docker install

**Information about the Cluster**
- Kubernetes version: v1.20.7+k3s1
- Cluster Type (Local/Downstream): Local

**Describe the bug**
Download KubeConfig fails to download when MCM is disabled

**To Reproduce**
- Deploy a single node docker install of rancher.
```
docker run -d --privileged --restart=unless-stopped -p 80:80 -p 443:443 rancher/rancher:master-head --features=multi-cluster-management=false
```
- Go to the cluster explorer of the local cluster
- Click on Download kubeconfig icon top right of the page
- Respoonse: `404 page not found`**

**Comment by StrongMonkey** [comment]
This is a duplicate issue of https://github.com/rancher/rancher/issues/33711. Can be retested or closed.

**Comment by catastrophe-brandon** [comment]
This worked in the local cluster with commit `d8824a1` on MacOS Chrome.

**Comment by StrongMonkey** [comment]
@catastrophe-brandon I don't think this has worked yet. Did you disable `multi-cluster-management=false`?

**Comment by catastrophe-brandon** [comment]
That was the part I forgot. Sorry for the confusion; reopening.

**Comment by catastrophe-brandon** [comment]
A minor clarification on the results described in the issue: the 404 surfaces in the browser's dev console. In the UI nothing happens when I click on the icon. Here's the error:

[riker.xyz-1627338075594.log](https://github.com/rancher/rancher/files/6881522/riker.xyz-1627338075594.log)

**Comment by StrongMonkey** [comment]
The explorer dashboard is relying on `/v3/clusters` api to generate kubeconfig, which is no longer available when running rancher with multi-cluster-management=false. https://github.com/rancher/rancher/pull/33811 introduce the same api on `v1/management.cattle.io.clusters` which is steve api(used by dashboard. This will require UI to switch to v1 api.

**Comment by StrongMonkey** [comment]
Above change is done in backend. Now UI can request kubeconfig from v1 steve api. For example `/v1/management.cattle.io.clusters/local?action=generateKubeconfig`
